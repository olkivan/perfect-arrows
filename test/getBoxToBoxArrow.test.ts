import { getBoxToBoxArrow } from "../src"

describe("getBoxToBoxArrow", () => {
  test("it computes an arrow between boxes", () => {
    expect(getBoxToBoxArrow(10, 10, 10, 10, 30, 30, 10, 10)).toMatchSnapshot()
  })

  test("it computes an arrow from a large box to a small box", () => {
    expect(
      getBoxToBoxArrow(10, 10, 576, 384, 600, 30, 10, 10)
    ).toMatchSnapshot()
  })

  test("it computes an arrow from a box back to itself", () => {
    expect(getBoxToBoxArrow(10, 10, 10, 10, 10, 10, 10, 10)).toMatchSnapshot()
  })

  test("it computes an arrow between boxes with straights disabled", () => {
    expect(
      getBoxToBoxArrow(10, 10, 10, 10, 30, 30, 10, 10, { straights: false })
    ).toMatchSnapshot()
  })
})
