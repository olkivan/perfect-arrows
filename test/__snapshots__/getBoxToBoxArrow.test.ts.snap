// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getBoxToBoxArrow it computes an arrow between boxes 1`] = `
Array [
  20,
  20,
  19.86019416763081,
  15.99767020863824,
  15.857864376269049,
  15.857864376269049,
  -3.1066757381017216,
  1.535879411306825,
  0.7853981633974483,
]
`;

exports[`getBoxToBoxArrow it computes an arrow between boxes with straights disabled 1`] = `
Array [
  20,
  20,
  19.86019416763081,
  15.99767020863824,
  15.857864376269049,
  15.857864376269049,
  -3.1066757381017216,
  1.535879411306825,
  0.7853981633974483,
]
`;

exports[`getBoxToBoxArrow it computes an arrow from a box back to itself 1`] = `
Array [
  10,
  15,
  -2.388680238973927e-15,
  5.000000000000002,
  -10,
  15.000000000000004,
  2.356194490192345,
  0.7853981633974482,
  0,
]
`;

exports[`getBoxToBoxArrow it computes an arrow from a box back to itsself 1`] = `
Array [
  10,
  15,
  -2.388680238973927e-15,
  5.000000000000002,
  -10,
  15.000000000000004,
  2.356194490192345,
  0.7853981633974482,
  0,
]
`;

exports[`getBoxToBoxArrow it computes an arrow from a large box to a small box 1`] = `
Array [
  586,
  45.33550488599349,
  583.781188639714,
  46.54248044680054,
  581.5623772794279,
  47.74945600760759,
  2.6433877446128227,
  -0.49820490897697056,
  -0.49820490897697733,
]
`;
